File: src/main/java/com/uiptv/model/PlayerResponse.java
package com.uiptv.model;

import lombok.Data;

@Data
public class PlayerResponse {
    private String url;
    private String drmType;
    private String drmLicenseUrl;
    private String clearKeysJson;
    private String inputstreamaddon;
    private String manifestType;

    public PlayerResponse(String url) {
        this.url = url;
    }

    public void setFromChannel(Channel channel) {
        this.drmType = channel.getDrmType();
        this.drmLicenseUrl = channel.getDrmLicenseUrl();
        this.clearKeysJson = channel.getClearKeysJson();
        this.inputstreamaddon = channel.getInputstreamaddon();
        this.manifestType = channel.getManifestType();
    }

    public void setFromBookmark(Bookmark bookmark) {
        this.drmType = bookmark.getDrmType();
        this.drmLicenseUrl = bookmark.getDrmLicenseUrl();
        this.clearKeysJson = bookmark.getClearKeysJson();
        this.inputstreamaddon = bookmark.getInputstreamaddon();
        this.manifestType = bookmark.getManifestType();
    }
}

File: server/src/main/kotlin/com/uiptv/model/PlayerResponse.kt
package com.uiptv.model

class PlayerResponse(url: String) {
    var url: String? = url
    var drmType: String? = null
    var drmLicenseUrl: String? = null
    var clearKeysJson: String? = null
    var inputstreamaddon: String? = null
    var manifestType: String? = null
    var ffmpegMode: String? = null
    var account: Account? = null
    var channel: Channel? = null

    fun setFromChannel(channel: Channel?, account: Account?) {
        this.channel = channel
        if (channel == null) {
            clearDrmMetadata()
            return
        }
        this.account = account
        drmType = channel.drmType
        drmLicenseUrl = channel.drmLicenseUrl
        clearKeysJson = channel.clearKeysJson
        inputstreamaddon = channel.inputstreamaddon
        manifestType = channel.manifestType
    }

    fun setFromBookmark(bookmark: Bookmark?, account: Account?) {
        if (bookmark == null) {
            clearDrmMetadata()
            return
        }
        this.account = account
        drmType = bookmark.drmType
        drmLicenseUrl = bookmark.drmLicenseUrl
        clearKeysJson = bookmark.clearKeysJson
        inputstreamaddon = bookmark.inputstreamaddon
        manifestType = bookmark.manifestType
    }

    private fun clearDrmMetadata() {
        drmType = null
        drmLicenseUrl = null
        clearKeysJson = null
        inputstreamaddon = null
        manifestType = null
    }
}
